### PR TITLE
[AzureMonitorExporter] refactor how EnvironmentVariables are read

### PR DESCRIPTION
**Pull request by TimothyMothra**
Change our Platform abstraction to read all environment variables at once.
Environment Variables are only read as part of sdk initialization.
After initialization, the platform object is disposed.

This also protects us from needing to use try/catch every time we get an environment variable.

## Changes
- `DefaultPlatform` will now collect all environment variables at once.
- remove `public IDictionary GetEnvironmentVariables();` from `IPlatform`.
  This is no longer needed.
- move `EnvironmentVariableConstants` into `Platform` directory. 
  One will never be used without the other.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Platform/DefaultPlatform.cs [hunk 1]
```diff
@@ -12,9 +12,9 @@ namespace Azure.Monitor.OpenTelemetry.Exporter.Internals.Platform
 {
     internal class DefaultPlatform : IPlatform
     {
-        public string? GetEnvironmentVariable(string name) => Environment.GetEnvironmentVariable(name);
+        private readonly IDictionary _environmentVariables = Environment.GetEnvironmentVariables();
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Platform/DefaultPlatform.cs:15]: Is this the correct approach? Why do we need to retain environment variables that are unrelated to the exporter?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Platform/DefaultPlatform.cs:15]: They aren't retained indefinitely. This is only part of the startup/initialization.

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Platform/DefaultPlatform.cs:15]: This can throw as well

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Platform/DefaultPlatform.cs]: Updated to not throw.

**Comment by TimothyMothra** [comment]
Debugging an unrelated issue and found this:

image

My changes in this PR would help avoid unnecessary exceptions.

**Review by rajkumar-rangaraj — APPROVED** [review]
